src/Relations/MorphToMany.php

```diff
@@ -0,0 +1,407 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Relations;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\MorphToMany as EloquentMorphToMany;
+use Illuminate\Support\Arr;
+
+use function array_diff;
+use function array_key_exists;
+use function array_keys;
+use function array_map;
+use function array_merge;
+use function array_reduce;
+use function array_values;
+use function count;
+use function is_array;
+use function is_numeric;
+
+class MorphToMany extends EloquentMorphToMany
+{
+    /**
+     * Get the key for comparing against the parent key in "has" query.
+     *
+     * @return string
+     */
+    public function getHasCompareKey()
+    {
+        return $this->foreignPivotKey;
+    }
+
+    /** @inheritdoc */
+    public function getRelationExistenceQuery(Builder $query, Builder $parentQuery, $columns = ['*'])
+    {
+        return $query;
+    }
```

Comment:
I think the parent method is fine and we don't need to override this. I will write test for this too.

Reply:
This one is related to `HybridRelations` and never called in my tests.